### PR TITLE
Feat/crud my user profile

### PR DESCRIPTION
**Pull request by pikaden**
click 
image
to go to profile

Author can view their profile, update account, change preference (change locale)

If they view other's profile, they cannot do that.
image

Added link to user's profile in search global and view quizbank

### CODE REVIEW

File: src/app/[locale]/(main)/profile/[id]/components/locale-switcher-select.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <p className="sr-only">{label}</p>
+      <select
+        className="inline-flex appearance-none bg-transparent py-3 pl-2 pr-6"
+        defaultValue={defaultValue}
+        disabled={isPending}
+        onChange={onSelectChange}
+      >
+        {children}
+      </select>
+    </label>
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/profile/[id]/components/locale-switcher-select.tsx:44]: https://ui.shadcn.com/docs/components/select

Use `<Select>` and `<Label>`

File: src/app/[locale]/(main)/profile/[id]/components/navigation-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  const t = useTranslations("Settings")
+  const [selectedId, setSelectedId] = useState("2")
+
+  const handleNavigationClick = useCallback((id: string) => {
+    setSelectedId(id)
+  }, [])
+
+  return (
+    <>
+      <div className="ml-20 mt-5 fixed w-16 flex-col items-start self-start rounded-lg border border-solid border-neutral-300 bg-neutral-50 py-3 pl-3 shadow-md max-md:ml-2.5">
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/profile/[id]/components/navigation-menu.tsx:67]: use `gap` to create space for children.

**Review by blackmouse572 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(main)/profile/[id]/components/navigation-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  handleClick,
+  nextSelectId,
+}: TooltipProps) => {
+  return (
+    <TooltipProvider>
+      <Tooltip>
+        <TooltipTrigger asChild>
+          <Button
+            variant={"ghost"}
+            className="mt-2.5 flex h-10 w-10/12 justify-center gap-2.5 rounded-xl p-1.5 shadow-sm"
```
[the remark this answers is not in the document]
Comment (pikaden) [reply src/app/[locale]/(main)/profile/[id]/components/navigation-menu.tsx]: Updated: deleted and change to NamedTooltip

File: src/app/[locale]/(main)/profile/[id]/profile/actions/fetch-user-profile.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { getToken } from "@/lib/auth"
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/profile/[id]/profile/actions/fetch-user-profile.tsx:1]: all action server must be mark as "use server" and function have "action" as suffix (ex: `getUserProfileAction()`)

File: src/app/[locale]/(main)/profile/[id]/profile/actions/update-user-profile.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import { UpdateProfileForm } from "@/types/update-profile-form"
+
+type UpdateProfileSchema = {
+  id: string
+  token: string
+  values: UpdateProfileForm
+}
+
+export const updateUserProfile = async ({
+  id,
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/profile/[id]/profile/actions/update-user-profile.tsx:11]: all action server must be mark as "use server" and function have "action" as suffix (ex: getUserProfileAction())

File: src/app/[locale]/(main)/profile/[id]/account/actions/change-password.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+type ChangePasswordSchema = {
+  token: string
+  values: {
+    oldPassword: string
+    password: string
+  }
+}
+
+export const ChangePassword = async ({
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/profile/[id]/account/actions/change-password.tsx:11]: all action server must be mark as "use server" and function have "action" as suffix (ex: getUserProfileAction())

**Review by blackmouse572 — CHANGES_REQUESTED** [review]
all action server must be mark as "use server" and function have "action" as suffix (ex: getUserProfileAction())

**Review by blackmouse572 — APPROVED** [review]

**Pull request merged**